operator/builtin/output/otlp/converter.go

```diff
@@ -0,0 +1,244 @@
+package otlp
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/observiq/stanza/entry"
+	"github.com/observiq/stanza/version"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+// Convert converts a slice of entries to pdata.Logs format
+func Convert(entries []*entry.Entry) pdata.Logs {
+
+	out := pdata.NewLogs()
+	logs := out.ResourceLogs()
+
+	entriesByResource := groupByResource(entries)
+
+	logs.Resize(len(entriesByResource))
+
+	for i, resourceEntries := range entriesByResource {
+		rls := logs.At(i)
+		resource := rls.Resource()
+		resource.InitEmpty()
+
+		resourceAtts := resource.Attributes()
+		for k, v := range resourceEntries[0].Resource {
+			resourceAtts.InsertString(k, v)
+		}
+
+		rls.InstrumentationLibraryLogs().Resize(1)
+		ills := rls.InstrumentationLibraryLogs().At(0)
+		ills.InitEmpty()
+
+		il := ills.InstrumentationLibrary()
+		il.InitEmpty()
+		il.SetName("stanza")
+		il.SetVersion(version.GetVersion())
+
+		for _, entry := range resourceEntries {
+			lr := pdata.NewLogRecord()
+			lr.InitEmpty()
+			lr.SetTimestamp(pdata.TimestampUnixNano(entry.Timestamp.UnixNano()))
+
+			sevText, sevNum := convertSeverity(entry.Severity)
+			lr.SetSeverityText(sevText)
+			lr.SetSeverityNumber(sevNum)
+
+			if len(entry.Labels) > 0 {
+				attributes := lr.Attributes()
+				for k, v := range entry.Labels {
+					attributes.InsertString(k, v)
+				}
+			}
+
+			lr.Body().InitEmpty()
+			insertToAttributeVal(entry.Record, lr.Body())
+
+			ills.Logs().Append(lr)
+		}
+	}
+
+	return out
+}
+
+func groupByResource(entries []*entry.Entry) [][]*entry.Entry {
+	resourceMap := make(map[string][]*entry.Entry)
+
+	for _, ent := range entries {
+		resourceBytes, err := json.Marshal(ent.Resource)
```

Comment:
We could save a bunch of allocations and memory if we reuse a JSON encoder that writes to a hasher (https://golang.org/pkg/hash/). I know you mentioned performance concerns, so this may help.

Reply:
You ok with leaving it for now? Seems like an optimization we can add if necessary.

Reply:
Yep, fine with me